src/pytorch_lightning/strategies/sharded_spawn.py

```diff
@@ -40,11 +40,19 @@ class DDPSpawnShardedStrategy(DDPSpawnStrategy):
 
     strategy_name = "ddp_sharded_spawn"
 
+    def connect(self, model: "pl.LightningModule") -> None:
+        if not _FAIRSCALE_AVAILABLE:  # pragma: no cover
+            raise MisconfigurationException(
+                "`DDPSpawnShardedStrategy` requires `fairscale` to be installed."
+                " Install it by running `pip install fairscale`."
+            )
```

Comment:
this should be in init

Reply:
You can see below, it was originally in the property. I'm moving this to connect as this was the closest to the property in terms of accessing the module. I won't move it to init to keep the behavior as compatible as possible.

Reply:
okay, but it should be in init.. let's ignore this for now! will do a follow-up to move it.